server/Arcadia.Assistant.Web/Download/DownloadActor.cs

```diff
@@ -0,0 +1,142 @@
+﻿namespace Arcadia.Assistant.Web.Download
+{
+    using System;
+    using System.Linq;
+    using System.Net.Http;
+    using System.Threading.Tasks;
+
+    using Microsoft.AspNetCore.Hosting;
+
+    using Akka.Actor;
+    using Akka.Event;
+
+    using Arcadia.Assistant.Web.Configuration;
+
+    public class DownloadActor : UntypedActor, ILogReceive, IWithUnboundedStash
+    {
+        private readonly ILoggingAdapter logger = Context.GetLogger();
+
+        private readonly IActorRef androidDownloadBuildActor;
+        private readonly IActorRef iosDownloadBuildActor;
+
+        public DownloadActor(
+            IDownloadApplicationSettings downloadApplicationSettings,
+            IHttpClientFactory httpClientFactory,
+            IHostingEnvironment hostingEnvironment)
+        {
+            this.androidDownloadBuildActor = Context.ActorOf(
+                Props.Create(() => new DownloadApplicationActor(
+                    downloadApplicationSettings,
+                    httpClientFactory,
+                    hostingEnvironment,
+                    downloadApplicationSettings.AndroidGetBuildsUrl,
+                    downloadApplicationSettings.AndroidGetBuildDownloadLinkTemplateUrl)),
+                "download-android-build");
+
+            this.iosDownloadBuildActor = Context.ActorOf(
+                Props.Create(() => new DownloadApplicationActor(
+                    downloadApplicationSettings,
+                    httpClientFactory,
+                    hostingEnvironment,
+                    downloadApplicationSettings.IosGetBuildsUrl,
+                    downloadApplicationSettings.IosGetBuildDownloadLinkTemplateUrl)),
+                "download-ios-build");
+
+            Context.System.Scheduler.ScheduleTellRepeatedly(
+                TimeSpan.Zero,
+                TimeSpan.FromMinutes(downloadApplicationSettings.DownloadBuildIntervalMinutes),
+                this.Self,
+                RefreshApplicationBuilds.Instance,
+                this.Self);
+        }
+
+        public IStash Stash { get; set; }
+
+        protected override void OnReceive(object message)
+        {
+            switch (message)
+            {
+                case GetLatestApplicationBuildPath msg:
+                    this.RespondLatestApplicationBuildPath(msg);
+                    break;
+
+                case RefreshApplicationBuilds msg:
+                    this.Self.Tell(msg);
+                    this.BecomeStacked(this.DownloadingApplicationBuilds);
+                    break;
+
+                default:
+                    this.Unhandled(message);
+                    break;
+            }
+        }
+
+        private void DownloadingApplicationBuilds(object message)
+        {
+            switch (message)
+            {
+                case RefreshApplicationBuilds _:
```

Comment:
bad. you might receive RefreshApplicationBuilds message while still in a process of refreshing.

Reply:
Fixed.